libsrc/vic20/tgi/vic20-hi.s

```diff
@@ -0,0 +1,1054 @@
+; SETDRAWPAGE: Set the drawable page. Called with the new page in A (0..n).
+; The page number is already checked to be valid by the graphics kernel.
+;
+; Must set an error code: NO (will only be called if page ok)
+;
+
+.proc SETDRAWPAGE
+        rts
+.endproc
+
+; ------------------------------------------------------------------------
+; SETCOLOR: Set the drawing color (in A). The new color is already checked
+; to be in a valid range (0..maxcolor-1).
+;
+; Must set an error code: NO (will only be called if color ok)
+;
+
+.proc SETCOLOR
+        sta     CURCOL
+        tax
+        beq     @L1
+        lda     #$FF
+@L1:    sta     BITMASK
+        rts
+.endproc
+
+; ------------------------------------------------------------------------
+; SETPALETTE: Set the palette (not available with all drivers/hardware).
+; A pointer to the palette is passed in ptr1. Must set an error if palettes
+; are not supported
+;
+; Must set an error code: YES
+;
+
+.proc SETPALETTE
+        ldy     #PALETTESIZE - 1
+@L1:    lda     (ptr1),y        ; Copy the palette
+        and     #$0F            ; Make a valid color
+        sta     PALETTE,y
+        dey
+        bpl     @L1
+
+; Initialize the color map with the new color settings.
+
+        ldy     #$00
+        lda     PALETTE+1       ; Foreground color
+@L2:    sta     CBASE+$0000,y
+        sta     CBASE+$0100,y
+        iny
+        bne     @L2
+        lda     $900f
+        and     #$f8
+        ora     PALETTE         ; Background color
+        sta     $900f
+
+; Done, reset the error code
+
+        lda     #TGI_ERR_OK
+        sta     ERROR
+        rts
+.endproc
+
+; ------------------------------------------------------------------------
+; GETPALETTE: Return the current palette in A/X. Even drivers that cannot
+; set the palette should return the default palette here, so there's no
+; way for this function to fail.
+;
+; Must set an error code: NO
+;
+
+.proc GETPALETTE
+        lda     #<PALETTE
+        ldx     #>PALETTE
+        rts
+.endproc
+
+; ------------------------------------------------------------------------
+; GETDEFPALETTE: Return the default palette for the driver in A/X. All
+; drivers should return something reasonable here, even drivers that don't
+; support palettes, otherwise the caller has no way to determine the colors
+; of the (not changeable) palette.
+;
+; Must set an error code: NO (all drivers must have a default palette)
+;
+
+.proc GETDEFPALETTE
+        lda     #<DEFPALETTE
+        ldx     #>DEFPALETTE
+        rts
+.endproc
+
+; ------------------------------------------------------------------------
+; SETPIXEL: Draw one pixel at X1/Y1 = ptr1/ptr2 with the current drawing
+; color. The coordinates passed to this function are never outside the
+; visible screen area, so there is no need for clipping inside this function.
+;
+; Must set an error code: NO
+;
+
+.proc SETPIXEL
+        jsr     CALC            ; Calculate coordinates
+
+        ldy     #$00
```

Comment:
Sometimes you write `#0`, sometimes `#00` and sometimes `#$00`. Does this denote some specific difference?

Reply:
No, just like it terse. Just another second.